drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -9015,10 +9049,11 @@ void RenderingDeviceVulkan::initialize(VulkanContext *p_context, bool p_local_de
 		vmaCreateAllocator(&allocatorInfo, &allocator);
 	}
 
+	const uint32_t frame_count = context->get_frame_count();
```

Comment:
When using a local rendering device we will end up allocating resources for many frames even though we only need one.

Reply:
You're thinking this in terms of "visual frames" but frame_count what really does is tracking synchronization boundaries.

If we only have one internal "frame", then after every submit we must stall.

Think of CPU and GPU as two horses racing in circles, and the GPU horse must ALWAYS stay behind the CPU horse.
And the CPU horse must NEVER overtake the GPU horse by more than N laps (N = frame_count).
But the GPU horse can only start galloping when the CPU horse raider says "submit".

Even if the user never ever sees a single window, we need a way to prevent the CPU horse from overtaking the GPU horse. If frame_count = 1, that's basically the same as the CPU horse saying "submit" and then hard-breaking until the GPU horse catches up.